src/main/java/seedu/algobase/model/searchrule/problemsearchrule/ProblemSearchRule.java

```diff
@@ -39,10 +41,29 @@ public boolean isSameProblemSearchRule(ProblemSearchRule other) {
             && other.getName().equals(getName());
     }
 
+    /**
+     * Returns true if two {@code Optional} are both present and have the same content or are both empty.
+     * @param first optional object
+     * @param second optional object
+     */
+    private boolean optionalEquals(Optional first, Optional second) {
```

Comment:
This can be refactored to be a util method

Reply:
I added it to `AppUtil`, thanks!